copy/copy.go

```diff
@@ -918,6 +1091,32 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 	if canModifyBlob && !isConfig {
 		uploadedInfo.CompressionAlgorithm = &desiredCompressionFormat
 	}
+	if decrypted {
+		uploadedInfo.CryptoOperation = types.Decrypt
+
+		// Ensure that encryption annotations are present in the new manifest
+		// to enable check authorization at a later time.
+		// TODO: Copy over only encryption related annotations
+		if uploadedInfo.Annotations == nil {
+			uploadedInfo.Annotations = map[string]string{}
+		}
+		for k, v := range srcInfo.Annotations {
+			uploadedInfo.Annotations[k] = v
+		}
+	}
+	if encrypted {
```

Comment:
Micro optimization: `} else if encrypted {`

Reply:
I believe it is possible to do decryption and re-encrypting an image in a single pass of `copy.Image`. So i think this should be okay?

Reply:
But in that case, let me add a statement to set it to PreserveOriginal.

Reply:
> I believe it is possible to do decryption and re-encrypting an image in a single pass of copy.Image. So i think this should be okay?

Very fair point! We should add a test to Skopeo for that.

Reply:
I discussed this with @harche and seems like for now it is more straightforward to do either encryption or decryption in a single pass - there isn't a usecase to how it is being done now to do both at the same time. I'll do some checking to ensure that in the library as well.